tests/v3/test_v2.py

```diff
@@ -44,7 +44,7 @@ def test_simple(store: StorePath) -> None:
         ("float64", 0.0),
         ("|S1", b""),
         ("|U1", ""),
-        ("object", 0),
+        ("object", ""),
```

Comment:
I think we agreed that this is the behavior we want. zarr-python 2.x only halfway special cased `dtype=str`. It's implicit fill value when you do `zarr.create(..., dtype=str)` was still the default for `object` (`0`).

```python
In [1]: import zarr  # 2.18.3
In [2]: arr = zarr.create(shape=10, dtype="str")

In [3]: arr.dtype
Out[3]: dtype('O')

In [4]: arr[:]
Out[4]: array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=object)
```

Reply:
This looks much better!